File: seed/csharp-model/grpc-proto-exhaustive/no-custom-config/src/SeedApi/DescribeResponse.cs
using System.Text.Json.Serialization;
using SeedApi.Core;
using Proto = Data.V1.Grpc;

#nullable enable

namespace SeedApi;

public record DescribeResponse
{
    [JsonPropertyName("namespaces")]
    public Dictionary<string, NamespaceSummary>? Namespaces { get; set; }

    [JsonPropertyName("dimension")]
    public uint? Dimension { get; set; }

    [JsonPropertyName("fullness")]
    public float? Fullness { get; set; }

    [JsonPropertyName("totalCount")]
    public uint? TotalCount { get; set; }

    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }

    /// <summary>
    /// Maps the DescribeResponse type into its Protobuf-equivalent representation.
    /// </summary>
    internal Proto.DescribeResponse ToProto()
    {
        var result = new Proto.DescribeResponse();
        if (Namespaces != null && Namespaces.Any())
        {
            foreach (var kvp in Namespaces)
            {
                result.Namespaces.Add(kvp.Key, kvp.Value.ToProto());
            }
            ;
        }
        if (Dimension != null)
        {
            result.Dimension = Dimension ?? 0;
        }
        if (Fullness != null)
        {
            result.Fullness = Fullness ?? 0.0f;
        }
        if (TotalCount != null)
        {
            result.TotalCount = TotalCount ?? 0;
        }
        return result;
    }

    /// <summary>
    /// Returns a new DescribeResponse type from its Protobuf-equivalent representation.
    /// </summary>
    internal static DescribeResponse FromProto(Proto.DescribeResponse value)
    {
        return new DescribeResponse
        {
            Namespaces = value.Namespaces?.ToDictionary(
                kvp => kvp.Key,
                kvp => NamespaceSummary.FromProto(kvp.Value)
            ),
            Dimension = value.Dimension,
            Fullness = value.Fullness,
            TotalCount = value.TotalCount,
        };
    }
}

File: seed/php-sdk/grpc-proto-exhaustive/src/Types/DescribeResponse.php
<?php

namespace Seed\Types;

use Seed\Core\Json\JsonSerializableType;
use Seed\Core\Json\JsonProperty;
use Seed\Core\Types\ArrayType;

class DescribeResponse extends JsonSerializableType
{
    /**
     * @var ?array<string, NamespaceSummary> $namespaces
     */
    #[JsonProperty('namespaces'), ArrayType(['string' => NamespaceSummary::class])]
    public ?array $namespaces;

    /**
     * @var ?int $dimension
     */
    #[JsonProperty('dimension')]
    public ?int $dimension;

    /**
     * @var ?float $fullness
     */
    #[JsonProperty('fullness')]
    public ?float $fullness;

    /**
     * @var ?int $totalCount
     */
    #[JsonProperty('totalCount')]
    public ?int $totalCount;

    /**
     * @param array{
     *   namespaces?: ?array<string, NamespaceSummary>,
     *   dimension?: ?int,
     *   fullness?: ?float,
     *   totalCount?: ?int,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->namespaces = $values['namespaces'] ?? null;
        $this->dimension = $values['dimension'] ?? null;
        $this->fullness = $values['fullness'] ?? null;
        $this->totalCount = $values['totalCount'] ?? null;
    }
}
